spidermon/contrib/actions/slack/__init__.py

```diff
@@ -27,7 +27,7 @@ def __init__(self, sender_token=None, sender_name=None, fake=False):
             raise NotConfigured("You must provide a slack sender name.")
 
         self.fake = fake
-        self._client = SlackClient(sender_token)
+        self._client = WebClient(sender_token)
         self._users = None
 
     @property
```

Comment:
**thought**: We have [this PR](https://github.com/scrapinghub/spidermon/pull/260/) opened that changes error handling for `_get_users_info`. If we are going to change how the call is made (as I commented some lines below), maybe we should try to incorporate her changes to make sure that we will not have the issue or, if the code changes are too much, take care to solve the problem she mentioned.

Reply:
It doesn't seem to me that the changes will be too much... We can choose to go either of the two ways: I can incorporate her changes to this PR, or after this PR is merged, @karenyavine can update her PR. What do you think?

Reply:
I made some tests using the changes suggested by that PR. Those changes only work when Spidermon sends messages to a channel, but it will fail when sending messages to a single user (private messages), though. That's because to send a message we need the `user_id` that comes from the `users_list` method.

Reply:
I hadn't considered the bot messaging via DM (is it even used? I'd assume alerting via dm would be less popular?). But It think it’d be safest to ask for the `user_id` to be supplied as an config, in order to maintain the least amount of privs.

Reply:
It would be the same for a single user or a list of users. In that case, the Spidermon settings would only accept user_id(s) and not username(s) anymore. I agree the safest way is to use the `users_info` method instead of the `users_list`, but I can't see another solution

Reply:
I did some research on this topic and maybe we won't have a better solution other than iterating over the entire list of users until we find the one we want.

We allow messaging via DM and as we can define a list of users that will receive the message, this is a feature that we should keep providing. I looked at the internal codebase of Scrapinghub and found the the majority uses a channel, but there are a few using DM.

According to [Slack](https://api.slack.com/changelog/2017-09-the-one-about-usernames), they will not provide an easy way to get `user_id` base on the username (they want to get rid of usernames!), so the only solution for now will be continue [iterating over all users](https://github.com/slackapi/node-slack-sdk/issues/73#issuecomment-497127404) until we find the one we want. To be fair, this makes no sense to me!

Asking for the `user_id` in the config (instead of the username) will add an extra burden to the user, because it is not that easy (not programatically) to discover the `user_id` of a user.